src/email.rs

```diff
@@ -0,0 +1,173 @@
+use std::ffi::CString;
+use std::ptr;
+
+use crate::util::Binding;
+use crate::{raw, Buf, Commit, DiffFindOptions, DiffOptions, Error, IntoCString};
+use crate::{Diff, Oid, Signature};
+
+/// A structure to represent patch in mbox format for sending via email
+pub struct Email {
+    /// Buffer to store the e-mail patch in
+    pub buf: Buf,
+}
+
+/// Options for controlling the formatting of the generated e-mail.
+pub struct EmailCreateOptions {
+    subject_prefix: Option<CString>,
+    raw: raw::git_email_create_options,
+}
+
+impl Default for EmailCreateOptions {
+    fn default() -> Self {
+        // Defaults options created in corresponding to `GIT_EMAIL_CREATE_OPTIONS_INIT`
+        let mut diff_opts = DiffOptions::new();
+        diff_opts.show_binary(true).context_lines(3);
+        unsafe {
+            let default_options = raw::git_email_create_options {
+                version: raw::GIT_EMAIL_CREATE_OPTIONS_VERSION,
+                flags: raw::GIT_EMAIL_CREATE_DEFAULT as u32,
+                diff_opts: ptr::read(diff_opts.raw()),
+                diff_find_opts: ptr::read(DiffFindOptions::new().raw()),
```

Comment:
Instead of using unsafe code like `ptr::read` here I think this builder should probably embedd the builders for the other options? That way all the options can continue to be configurable

Reply:
makes sense, changed.